lib/http2.c

```diff
@@ -271,6 +271,16 @@ static CURLcode http2_data_setup(struct Curl_cfilter *cf,
   return CURLE_OK;
 }
 
+static void free_push_headers(struct h2_stream_ctx *stream)
+{
+  size_t i;
+  for(i = 0; i<stream->push_headers_used; i++)
+    free(stream->push_headers[i]);
+  free(stream->push_headers);
```

Comment:
nitpick, but wouldn't `Curl_safefree()` be more appropriate here?

Reply:
:+1:

Reply:
Changed.